packages/grid/x-data-grid-pro/src/components/DataGridProVirtualScroller.tsx

```diff
@@ -274,8 +328,75 @@ const DataGridProVirtualScroller = React.forwardRef<
 
   const detailPanels = getDetailPanels();
 
+  const topPinnedRows = getRows({ renderContext, rows: topPinnedRowsData });
+
+  const pinnedRowsHeight = calculatePinnedRowsHeight(apiRef);
+
+  const mainRows = getRows({
+    renderContext,
+    rowIndexOffset: topPinnedRowsData.length,
+  });
+
+  const bottomPinnedRows = getRows({
+    renderContext,
+    rows: bottomPinnedRowsData,
+    rowIndexOffset: topPinnedRowsData.length + (mainRows ? mainRows.length : 0),
+  });
+
+  const contentProps = getContentProps();
+
+  const pinnedColumnsStyle = { minHeight: contentProps.style.minHeight };
+
+  if (contentProps.style.minHeight && contentProps.style.minHeight === '100%') {
+    contentProps.style.minHeight = `calc(100% - ${pinnedRowsHeight.top}px - ${pinnedRowsHeight.bottom}px)`;
+  }
+
   return (
     <GridVirtualScroller {...getRootProps(other)}>
+      {topPinnedRowsData.length > 0 ? (
+        <VirtualScrollerPinnedRows
+          className={classes.topPinnedRows}
+          ownerState={{ position: 'top' }}
+          style={{ width: contentProps.style.width, height: pinnedRowsHeight.top }}
+        >
+          {leftRenderContext && (
+            <VirtualScrollerPinnedColumns
+              className={classes.leftPinnedColumns}
+              ownerState={{ side: GridPinnedPosition.left }}
+            >
+              {getRows({
+                renderContext: leftRenderContext,
+                minFirstColumn: leftRenderContext.firstColumnIndex,
+                maxLastColumn: leftRenderContext.lastColumnIndex,
+                availableSpace: 0,
+                ignoreAutoHeight: true,
+                rows: topPinnedRowsData,
+              })}
+            </VirtualScrollerPinnedColumns>
+          )}
+          <VirtualScrollerPinnedRowsRenderZone
+            className={classes.pinnedRowsRenderZone}
+            ref={topPinnedRowsRenderZoneRef}
```

Comment:
Same for the render zone of the bottom pinned rows.

```suggestion
            ref={topPinnedRowsRenderZoneRef}
            role="presentation"
```